README.md

```diff
@@ -797,6 +797,82 @@ In either case:
     end
     ```
 
+### Nested conditionals
+
+* <a name="no-nested-conditionals"></a>
+  Avoid the use of nested conditionals for flow of control. 
+  ([More on this][avoid-else-return-early].) <sup>[[link](#no-nested-conditionals)]</sup>
+
+  Prefer a guard clause when you can assert invalid data. A guard clause
+  is a conditional statement at the top of a function that bails out as
+  soon as it can. 
+  
+  The general principles boil down to:
+  * Return as soon as you know your method cannot do any more meaningful work
+  * Reduce indentation by using `if`/`unless`/`return` instead of a top-level 
+  `if`/`else`
+  * Try keep the “meat” of your method at the lowest indentation level.
+
+  ```ruby
+  # bad
+  def compute_thing(thing)
+    if thing[:foo]
+      update_with_bar(thing[:foo])
+      if thing[:foo][:bar]
+        partial_compute(thing)
+      else
+        re_compute(thing)
+      end
+    else
+      []
+    end
+  end
+
+  # good
+  def compute_thing(thing)
+    return [] unless thing[:foo]
+    update_with_bar(thing[:foo])
+    return re_compute(thing) unless thing[:foo][:bar]
+    partial_compute(thing)
+  end
+  ```
+
+  Prefer `next` in loops instead of conditional blocks.
+
+  ```ruby
+  # bad
+  [0, 1, 2, 3].each do |item|
+    if item > 1
+      puts item
+    end
+  end
+
+  # good
+  [0, 1, 2, 3].each do |item|
+    next unless item > 1
+    puts item
+  end
+  ```
+
+  The same technique can be used to untangle complex booleans. 
+  
+  ```ruby
+  # bad
+  def should_do_something?
+    condition_a && (((condition_b || condition_c) && condition_d) || condition_e)
```

Comment:
I don't feel this is a good example for refactoring towards guarding ifs. The result below with its mix of `false unless` and `true if` is just as unreadable. Complex booleans like this should rather assigned partial predicates to intention revealing local variables.
```
def should_do_something?
  user_has_been_ghosted = ((condition_b || condition_c) && condition_d)
  condition_a && (user_has_been_ghosted || condition_e)
end
```